### ISSUE TITLE
Don't regard `-{` and `}-` as language conversion within <math> tags

### ISSUE THREAD

**Issue by AlexanderMisel** [gnosygnu/xowa#107]
For 
```mediawiki
:<math>x_{1}-1</math>
:<math>x-{1+2}-1</math>
```
XOWA generates
```html
<dd>
  <span id="xowa_math_txt_0">x_{11</span>}-
</dd>
<dd>
  <span id="xowa_math_txt_0">x1+21</span>
</dd>
```
before MathJax is used, I think it's because it regarded `-{` and `}-` as language variant conversion, but actually language conversion is not applied to math equations. 

This often causes MathJax display error like `Extra open brace or missing close brace`. Please fix it.

**Comment by gnosygnu** [comment]
Thanks for the detailed error report.

I agree that this is a fairly high priority. However, I took a look at this, and the fix isn't easy. If I can't get it into this weekend's release, then it will be next week.

**Comment by markusmaresch** [comment]
This is probably related to Issue 10.
https://github.com/gnosygnu/xowa/issues/10

**Comment by gnosygnu** [comment]
Thanks for the reminder. Actually #10 is similar, but the underlying problem is something different.

* #107 has to do with multi-lingual code and special MediaWiki syntax to handle Chinese / Serbian languages
* #10 has to do with Math code and extra OCaml verification to allow lax input.

I know #10 has been outstanding for a while, but it involves a fairly major change to the Math system, and I've been hesitant. I'll look at it again in the coming weeks, but it's not easy.

Let me know if you're seeing #10 on more and more Wikipedia pages, and I'll definitely bump the priority up.

Thanks.

**Comment by gnosygnu** [comment]
@AlexanderMisel: It was a bit difficult, but I built a preview version that resolves this issue. You can download the Windows 64-bit version here: https://drive.google.com/open?id=0B9cb52zjL2rIaHVUN09WWVh1dTg . Let me know if you need a different Operating System, and I'll upload it.

I wrote an automated test for it, and also tested it directly with the app by using your example above. Let me know if you come across any pages (or samples) that don't work.

I really do appreciate the error report as well as the clean *before* and *after* test case. If it means anything, I can credit you in the XOWA change log for this report with some sort of nickname. Of course, it's perfectly ok to remain anonymous also. :)

@markusmaresch: I know #10 is old but the fix is complicated (I have to rewrite OCaml parsing code in Java). I'll take another look at it this month, and see if I can do anything else with it. I'll comment in #10 in either case.

Thanks.

**Comment by AlexanderMisel** [comment]
It works, thank you for fixing this problem.

**Comment by gnosygnu** [comment]
Cool! Thanks for the follow-up! I'll close this ticket out in a few days

**Comment by gnosygnu** [comment]
As this item is included in tonight's [v3.12.1.1](https://github.com/gnosygnu/xowa/releases) release, I'm marking this ticket closed. Let me know if there's anything else. Thanks.
